### PR TITLE
feat: update to findex 5

### PR DESCRIPTION
**Pull request by Manuthor**

### CODE REVIEW

**Review by tbrezot — COMMENTED** [review]
Did you test that adding an already existing keyword does not return anything?

File: lib/src/findex/findex.dart [hunk 1]
```diff
@@ -115,8 +125,21 @@ class Findex {
       );
       final end = DateTime.now();
 
+      if (errorCode != 0 &&
+          outputLengthPointer.value > defaultOutputSizeInBytes) {
+        return upsert(masterKey, label, additions, deletions, fetchEntries,
+            upsertEntries, insertChains,
+            outputSizeInBytes: outputLengthPointer.value);
+      }
```
Comment (tbrezot) [inline lib/src/findex/findex.dart]: You should not upsert a second time.

**Review by tbrezot — CHANGES_REQUESTED** [review]

File: lib/src/findex/findex.dart (hunk 1 above)
Comment (Manuthor) [reply lib/src/findex/findex.dart]: as discussed (and after a change in cloudproof_rust), the first call to `h_upsert` is only used to get the allocation size to use in the 2nd call

**Comment by Manuthor** [comment]
> Did you test that adding an already existing keyword does not return anything?

I've just added a dedicated test on upsert. Thanks for the suggestion.

**Pull request merged**